src/frontend/xgboost_json.cc

```diff
@@ -25,45 +43,358 @@ namespace frontend {
+      sizeof(readBuffer));
+  *out = std::move(ParseStream(std::move(input_stream)));
+  fclose(fp);
 }
 
-void LoadXGBoostJSONModel(std::string json_str, Model* out) {
-  dmlc::MemoryStringStream fs(&json_str);
-  *out = std::move(ParseStream(&fs));
+void LoadXGBoostJSONModelString(const char* json_str,
+                                size_t length, Model *out) {
+  auto input_stream = std::make_unique<rapidjson::MemoryStream>(
+      json_str, length);
+  *out = std::move(ParseStream(std::move(input_stream)));
 }
 
 }  // namespace frontend
-}  // namespace treelite
 
-namespace {
+namespace details {
+
+/******************************************************************************
+ * BaseHandler
+ * ***************************************************************************/
+
+bool BaseHandler::pop_handler() {
+  if (auto parent = delegator.lock()) {
+    parent->pop_delegate();
+    return true;
+  } else {
+    return false;
+  }
+}
+
+void BaseHandler::set_cur_key(const char *str, std::size_t length) {
+  cur_key = std::string{str, length};
+}
+
+const std::string &BaseHandler::get_cur_key() { return cur_key; }
+
+bool BaseHandler::check_cur_key(const std::string &query_key) {
+  return cur_key == query_key;
+}
+
+template <typename ValueType>
+bool BaseHandler::assign_value(const std::string &key,
+                               ValueType &&value,
+                               ValueType &output) {
+  if (check_cur_key(key)) {
+    output = value;
+    return true;
+  } else {
+    return false;
+  }
+}
+
+template <typename ValueType>
+bool BaseHandler::assign_value(const std::string &key,
+                  const ValueType &value,
+                  ValueType &output) {
+  return assign_value(key, std::forward<ValueType>(value), output);
+}
+
+/******************************************************************************
+ * IgnoreHandler
+ * ***************************************************************************/
+bool IgnoreHandler::Null() { return true; }
+bool IgnoreHandler::Bool(bool b) { return true; }
+bool IgnoreHandler::Int(int i) { return true; }
+bool IgnoreHandler::Uint(unsigned u) { return true; }
+bool IgnoreHandler::Int64(int64_t i) { return true; }
+bool IgnoreHandler::Uint64(uint64_t u) { return true; }
+bool IgnoreHandler::Double(double d) { return true; }
+bool IgnoreHandler::String(const char *str, std::size_t length, bool copy) {
+  return true; }
+bool IgnoreHandler::StartObject() { return push_handler<IgnoreHandler>(); }
+bool IgnoreHandler::Key(const char *str, std::size_t length, bool copy) {
+  return true; }
+bool IgnoreHandler::StartArray() { return push_handler<IgnoreHandler>(); }
+
+/******************************************************************************
+ * TreeParamHandler
+ * ***************************************************************************/
+bool TreeParamHandler::String(const char *str, std::size_t length, bool copy) {
+  // Key "num_deleted" not documented in schema
```

Comment:
`num_deleted` is an obsolete field which has been deprecated and removed.

Reply:
Interestingly, it's still being output by xgboost as far as I can tell. I'll update the comment to mention that it's deprecated. Thanks!